euphoria-core/src/main/java/cz/seznam/euphoria/core/client/operator/MapElements.java

```diff
@@ -136,6 +165,41 @@ public static OfBuilder named(String name) {
     return DAG.of(
         // do not use the client API here, because it modifies the Flow!
         new FlatMap<IN, OUT>(getName(), getFlow(), input,
-            (i, c) -> c.collect(mapper.apply(i)), null));
+            (i, c) -> c.collect(mapper.apply(i, new CollectorAdapter(c))), null));
```

Comment:
I think we can safely just cast the `Collector` to `Context` here, because if the user casts it back, then he knows he does something that is wrong.